server/data/prisonApiClient.ts

```diff
@@ -29,15 +37,140 @@ export default class PrisonApiClient extends AbstractHmppsRestClient {
+      authToken: user.token,
+    })
+  }
+
+  async getActivityList(
+    prisonId: string,
+    locationId: string,
+    date: string,
+    period: string,
+    usage: string,
+    user: ServiceUser,
+  ): Promise<PrisonerSchedule[]> {
+    return this.get({
+      path: `/api/schedules/${prisonId}/locations/${locationId}/usage/${usage}`,
+      query: { date, timeSlot: period },
+      authToken: user.token,
+    })
+  }
+
+  async getVisits(
+    prisonId: string,
+    date: string,
+    period: string,
+    offenderNumbers: string[],
+    user: ServiceUser,
+  ): Promise<ScheduledAppointmentDto[]> {
+    return this.post({
+      path: `/api/schedules/${prisonId}/visits`,
+      query: { date, timeSlot: period },
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  async getAppointments(
+    prisonId: string,
+    date: string,
+    period: string,
+    offenderNumbers: string[],
+    user: ServiceUser,
+  ): Promise<ScheduledAppointmentDto[]> {
+    return this.post({
+      path: `/api/schedules/${prisonId}/appointments`,
+      query: { date, timeSlot: period },
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  async getActivities(
+    prisonId: string,
+    date: string,
+    period: string,
+    offenderNumbers: string[],
+    user: ServiceUser,
+  ): Promise<ScheduledAppointmentDto[]> {
+    return this.post({
+      path: `/api/schedules/${prisonId}/activities`,
+      query: { date, timeSlot: period },
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  async getSentenceData(offenderNumbers: string[], user: ServiceUser): Promise<any[]> {
+    return this.post({
+      path: '/api/offender-sentences',
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  async getCourtEvents(
+    prisonId: string,
+    date: string,
+    offenderNumbers: string[],
+    user: ServiceUser,
+  ): Promise<CourtEvent[]> {
+    return this.post({
+      path: `/api/schedules/${prisonId}/courtEvents`,
+      query: { date },
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  async getExternalTransfers(
+    prisonId: string,
+    date: string,
+    offenderNumbers: string[],
+    user: ServiceUser,
+  ): Promise<TransferEvent[]> {
+    return this.post({
+      path: `/api/schedules/${prisonId}/externalTransfers`,
+      query: { date },
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  async getAlerts(prisonId: string, offenderNumbers: string[], user: ServiceUser): Promise<ScheduledAppointmentDto[]> {
+    return this.post({
+      path: `/api/bookings/offenderNo/${prisonId}/alerts`,
+      data: offenderNumbers,
+      authToken: user.token,
+    })
+  }
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  async getAssessments(code: string, offenderNumbers: string[], user: ServiceUser): Promise<any[]> {
```

Comment:
`Promise<Assessment[]>`

Reply:
As with the others - this types doesnt describe the actual response from the api. Not sure what to do here - I guess i could change the api test response fixtures to satisfy the type, other option is to create a new type but that seems just as wrong. Correct course of action might be to ask DPS to fix their api schema.

Reply:
As discussed yesterday i have created versions of the api types with very minor changes that are slightly more lenient - making a couple of attributes optional and changing enum types to strings. This has allowed me to replace the 'any' types.